docs/labs/hardcoded.js

```diff
@@ -7,48 +7,59 @@ info =
+      text_ja: "回答は元からあるように `conn = DriverManager.getConnection( url,` で始まるはずです。「リセット」ボタンを使うとよいかもしれません。",
     },
     {
       absent: String.raw`System \. getenv`,
       text: "Use `System.getenv` to retrieve an environment variable value. For example, use `System.getenv(\"USERNAME\")` to retrieve the username.",
+      text_ja: "環境変数の値を取得するためには `System.getenv` を使います。例えばユーザ名を取得するためには `System.getenv(\"USERNAME\")` を使ってください。",
     },
     {
       present: String.raw`System \. getenv \( \"admin\" \)`,
       text: "You need to pass to `System.getenv` the name of an environment variable value, not the result you might get. Do not use constructs like `System.getenv(\"admin\")`. Instead, for example, use `System.getenv(\"USERNAME\")` to retrieve the username.",
+      text_ja: "`System.getenv` には取得したい値ではなく、環境変数の名前を渡します。`System.getenv(\"admin\")` の代わりに、例えばユーザ名を取得するためには `System.getenv(\"USERNAME\")` としてください。",
     },
     {
       absent: String.raw`System \. getenv \( \"PASSWORD\" \)`,
       text: "Use `System.getenv` to retrieve an environment variable value. For example, use `System.getenv(\"USERNAME\")` to retrieve the username.",
+      text_ja: "環境変数の値を取得するためには `System.getenv` を使います。例えばユーザ名を取得するためには `System.getenv(\"USERNAME\")` を使ってください。",
     },
     {
       present: "admin",
       text: "The term 'admin' should not be in your code. You should be retrieving both the username and the password from somewhere else, in this case, from environment variables.",
+      text_ja: "'admin' という語句はコードに現れるべきではありません。ユーザ名とパスワードのどちらもどこか他の場所から、今回のケースでは環境変数の値から取得する必要があります。",
     },
     {
       present: "(system|Getenv|GetEnv)",
       text: "Java is case-sensitive. You need to use `System.getenv` and not some other variation of uppercase or lowercase.",
+      text_ja: "Java は大文字と小文字を区別します。`System.getenv` 以外の大文字と小文字の組み合わせを使ってはいけません。",
     },
     {
       absent: String.raw`\; \s* $`,
       text: "Java statements must end with a semicolon.",
+      text_ja: "Java のステートメントはセミコロンで終わる必要があります。",
     },
     {
       absent: String.raw`\) \) \; \s* $`,
       text: "Double-check your closing parentheses at the end of the statement.",
+      text_ja: "ステートメント終わりの右側（閉じ）カッコをもう一度確認してください。",
     },
     {
       present: String.raw`new\s+String`,
       text: "You do not need to construct a new string to retrieve an environment variable value.",
+      text_ja: "環境変数の値を取得するために新たに文字列を作成する必要はありません。",
     },
     {
       present: String.raw`^ conn = DriverManager \. getConnection \( url \) \; \s* $`,
       text: "In some sense this is correct, as long as the url is not hardcoded. However, it's often better if administrators can easily change the username or password separately, and it makes out point clearer. Please provide the username and password and separate values.",
+      text_ja: "URL がハードコードされていない限り、ある意味これは正解です。しかしユーザ名とパスワードを容易にかつ個別に変更できることはしばしばアドミニストレータにとって望ましく、かつそれにより問題の本質が明確になります。ユーザ名とパスワードは個別の値としてください。",
```

Comment:
```suggestion
      text_ja: "URL がハードコードされていない限り、ある意味これは正解です。しかしユーザ名とパスワードを容易にかつ個別に変更できることはしばしば管理者にとって望ましく、かつそれにより問題の本質が明確になります。ユーザ名とパスワードは個別の値としてください。",
```
Unless there's any specific intention to use "アドミニストレータ", "管理者" would be still a straightforward translation for "administrator".

Reply:
Thank you for your detailed review.
"administrator" = "管理者"
would be better.  Fixed it.